analyzers/src/SonarAnalyzer.CSharp/Rules/CognitiveComplexity.cs

```diff
@@ -37,6 +38,21 @@ public sealed class CognitiveComplexity : CognitiveComplexityBase<SyntaxKind>
 
         protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    if (c.ContainingSymbol is IMethodSymbol)
```

Comment:
Interesting take, I had to solve the same issue and didn't know how to determine if a `compilationUnit` is containing top level statements. Thanks 😄 

Could you please extract this to a method outside of this class so it can be reused?

Reply:
:) Thanks! Any ideas where should it belong to?

Reply:
Extension method (`IsTopLevelStatementsFile`) on the `SyntaxNodeAnalysisContext`?